src/gennylib/include/gennylib/conventions.hpp

```diff
@@ -164,6 +166,116 @@ namespace YAML {
 
 using genny::decodeNodeInto;
 
+template <>
+struct convert<mongocxx::read_preference> {
+    using ReadPreference = mongocxx::read_preference;
+    using ReadMode = mongocxx::read_preference::read_mode;
+    static Node encode(const ReadPreference& rhs) {
+        Node node;
+        auto mode = rhs.mode();
+        if (mode == ReadMode::k_primary) {
+            node["ReadMode"] = "primary";
+        } else if (mode == ReadMode::k_primary_preferred) {
+            node["ReadMode"] = "primaryPreferred";
+        } else if (mode == ReadMode::k_secondary) {
+            node["ReadMode"] = "secondary";
+        } else if (mode == ReadMode::k_secondary_preferred) {
+            node["ReadMode"] = "secondaryPreferred";
+        } else if (mode == ReadMode::k_nearest) {
+            node["ReadMode"] = "nearest";
+        }
+        auto maxStaleness = rhs.max_staleness();
+        if (maxStaleness) {
+            node["MaxStaleness"] = genny::TimeSpec(*maxStaleness);
+        }
+        return node;
+    }
+
+    static bool decode(const Node& node, ReadPreference& rhs) {
```

Comment:
@henrikingo Here's the code in genny to natively parse `mongocxx::read_preference` and `mongocxx::write_concern`. For read preference, it accepts the "ReadMode" and "MaxStaleness" top level keys. "ReadMode" accepts a number of strings as defined on lines 203 to 212.